app/javascript/shared/components/tags.jsx

```diff
@@ -87,144 +88,40 @@ class Tags extends Component {
-        <div className={`${this.props.classPrefix}__tagrules--${this.state.showingRulesForTag === tag.name ? 'active' : 'inactive'}`} dangerouslySetInnerHTML={{ __html: tag.rules_html }} />
-      </div>
-    ));
-    if (
-      this.state.searchResults.length > 0 &&
-      (document.activeElement.id === 'tag-input' || document.activeElement.className === 'articleform__tagsoptionrulesbutton')
-    ) {
-      searchResultsHTML = (
-        <div className={`${this.props.classPrefix}__tagsoptions`}>
-          {searchResultsRows}
-          <div className={`${this.props.classPrefix}__tagsoptionsbottomrow`}>Some tags have rules and guidelines determined by community moderators</div>
-        </div>
-      );
-    }
-
-    return (
-      <div className={`${this.props.classPrefix}__tagswrapper`}>
-        {this.props.listing && <label htmlFor="Tags">Tags</label>}
-        <input
-          id="tag-input"
-          type="text"
-          ref={t => (this.textArea = t)}
-          className={`${this.props.classPrefix}__tags`}
-          placeholder={`${this.props.maxTags} tags max, comma separated, no spaces or special characters`}
-          autoComplete='off'
-          value={this.props.defaultValue}
-          onInput={this.handleInput}
-          onKeyDown={this.handleKeyDown}
-          onBlur={this.handleFocusChange}
-          onFocus={this.handleFocusChange}
-        />
-        {searchResultsHTML}
-      </div>
-    );
-  }
-
-  handleRulesClick = e => {
-    e.preventDefault();
-    if (this.state.showingRulesForTag === e.target.dataset.content) {
-      this.setState({showingRulesForTag: null});
-    } else {
-      this.setState({showingRulesForTag: e.target.dataset.content});
-    }
-  }
-
-  handleTagClick = e => {
-    if (e.target.className === 'articleform__tagsoptionrulesbutton') {
-      return;
-    }
-    const input = document.getElementById('tag-input');
-    input.focus();
-    this.insertTag(e.target.dataset.content);
-  };
-
-  handleInput = e => {
-    let { value } = e.target;
-    // If we start typing immediately after a comma, add a space
-    // before what we typed.
-    // e.g. If value = "javascript," and we type a "p",
-    // the result should be "javascript, p".
-    if (e.inputType === 'insertText') {
-      const isTypingAfterComma =
-        e.target.value[e.target.selectionStart - 2] === ',';
-      if (isTypingAfterComma) {
-        value = this.insertSpace(value, e.target.selectionStart - 1);
-      }
-    }
-
-    if (e.data === ',') {
-      value += ' ';
-    }
-
-    this.props.onInput(value);
-
-    const query = this.getCurrentTagAtSelectionIndex(
-      e.target.value,
-      e.target.selectionStart - 1,
-    );
-
-    this.setState({
-      selectedIndex: 0,
-      cursorIdx: e.target.selectionStart,
-      prevLen: this.textArea.value.length,
-    });
-    return this.search(query);
-  };
-
-  getCurrentTagAtSelectionIndex(value, index) {
+  getCurrentTagAtSelectionIndex = (value, index) => {
```

Comment:
I opened the file and I see an error on line 128 because the inner function is not returning a value